tsl/test/expected/gapfill.out

```diff
@@ -1225,136 +1266,293 @@ SELECT
+ time_bucket_gapfill 
+---------------------
+                  -1
+                   0
+                   1
+                   2
+(4 rows)
+
 -- date time_bucket_gapfill with no start/finish
 SELECT
   time_bucket_gapfill('1d'::interval,t)
 FROM (VALUES ('1999-12-30'::date),('2000-01-01'::date)) v(t)
 WHERE
   t >= '1999-12-29' AND t < '2000-01-03'
 GROUP BY 1;
-ERROR:  invalid time_bucket_gapfill argument: start cannot be NULL
+ time_bucket_gapfill 
+---------------------
+ 12-29-1999
+ 12-30-1999
+ 12-31-1999
+ 01-01-2000
+ 01-02-2000
+(5 rows)
+
 -- timestamp time_bucket_gapfill with no start/finish
 SELECT
   time_bucket_gapfill('12h'::interval,t)
 FROM (VALUES ('1999-12-30'::timestamp),('2000-01-01'::timestamp)) v(t)
 WHERE
   t >= '1999-12-29' AND t < '2000-01-03'
 GROUP BY 1;
-ERROR:  invalid time_bucket_gapfill argument: start cannot be NULL
+   time_bucket_gapfill    
+--------------------------
+ Wed Dec 29 00:00:00 1999
+ Wed Dec 29 12:00:00 1999
+ Thu Dec 30 00:00:00 1999
+ Thu Dec 30 12:00:00 1999
+ Fri Dec 31 00:00:00 1999
+ Fri Dec 31 12:00:00 1999
+ Sat Jan 01 00:00:00 2000
+ Sat Jan 01 12:00:00 2000
+ Sun Jan 02 00:00:00 2000
+ Sun Jan 02 12:00:00 2000
+(10 rows)
+
 -- timestamptz time_bucket_gapfill with no start/finish
 SELECT
   time_bucket_gapfill('12h'::interval,t)
 FROM (VALUES ('1999-12-30'::timestamptz),('2000-01-01'::timestamptz)) v(t)
 WHERE
   t >= '1999-12-29' AND t < '2000-01-03'
 GROUP BY 1;
-ERROR:  invalid time_bucket_gapfill argument: start cannot be NULL
+     time_bucket_gapfill      
+------------------------------
+ Tue Dec 28 16:00:00 1999 PST
+ Wed Dec 29 04:00:00 1999 PST
+ Wed Dec 29 16:00:00 1999 PST
+ Thu Dec 30 04:00:00 1999 PST
+ Thu Dec 30 16:00:00 1999 PST
+ Fri Dec 31 04:00:00 1999 PST
+ Fri Dec 31 16:00:00 1999 PST
+ Sat Jan 01 04:00:00 2000 PST
+ Sat Jan 01 16:00:00 2000 PST
+ Sun Jan 02 04:00:00 2000 PST
+ Sun Jan 02 16:00:00 2000 PST
+(11 rows)
+
 -- timestamptz time_bucket_gapfill with more complex expression
 SELECT
   time_bucket_gapfill('12h'::interval,t)
 FROM (VALUES ('1999-12-30'::timestamptz),('2000-01-01'::timestamptz)) v(t)
 WHERE
   t >= '2000-01-03'::timestamptz - '4d'::interval AND t < '2000-01-03'
 GROUP BY 1;
-ERROR:  invalid time_bucket_gapfill argument: start cannot be NULL
+     time_bucket_gapfill      
+------------------------------
+ Wed Dec 29 16:00:00 1999 PST
+ Thu Dec 30 04:00:00 1999 PST
+ Thu Dec 30 16:00:00 1999 PST
+ Fri Dec 31 04:00:00 1999 PST
+ Fri Dec 31 16:00:00 1999 PST
+ Sat Jan 01 04:00:00 2000 PST
+ Sat Jan 01 16:00:00 2000 PST
+ Sun Jan 02 04:00:00 2000 PST
+ Sun Jan 02 16:00:00 2000 PST
+(9 rows)
+
+-- timestamptz time_bucket_gapfill with different datatype in finish constraint
+SELECT
+  time_bucket_gapfill('12h'::interval,t)
+FROM (VALUES ('1999-12-30'::timestamptz),('2000-01-01'::timestamptz)) v(t)
+WHERE
+  t >= '2000-01-03'::timestamptz - '4d'::interval AND t < '2000-01-03'::date
+GROUP BY 1;
```

Comment:
should we also have a test with `NOW()` in the where clause?

Reply:
added